src/core/libmaven/csvreports.cpp

```diff
@@ -199,10 +200,14 @@ void CSVReports::addGroup (PeakGroup* group) {
 
 }
 
-void CSVReports::insertPeakInformationIntoCSVFile(PeakGroup* group) {
-
-      writePeakInfo(group);
-
+void CSVReports::insertPeakInformationIntoCSVFile(PeakGroup* group)
+{
+    if (group->childCount() == 0) {
+        writePeakInfo(group);
+    } else {
+        for (auto& child : group->children)
+        writePeakInfo(&child);
```

Comment:
@saifulbkhan When a group CSV is created, we check which tracers have been selected during the time of export and only those labels are added to the CSV. Please make the behavior consistent between group and peak CSVs.

Reply:
I have made the change. Also the behavior you describe was disabled (`userSelectedIsotopesOnly` defaulted to false) always. I have enabled it. Let me know if this is not something that needs to happen right now.

Reply:
@saifulbkhan I'm not sure if only user selected isotopes should be exported or all detected labels should go in the CSV. Since tracer selection usually happens before peak detection, I can't picture why the user would go and deselect one tracer before exporting. Maybe you should consult the product team before making this call?